### PR TITLE
[Feat] Support non-gated activations in NVFP4 modelopt path

### PR DESCRIPTION
**Pull request by omera-nv**
## Purpose
For upcoming Nvidia releases, we require support for running non-gated activations for NVFP4 modelopt checkpoints.
This PR enables this by:
1. Modifying the sizes of the modelopt nvfp4 MoE weights according to the value of `self.moe.is_act_and_mul`
2. Calling the flashinfer cutlass fused moe kernel, which supports this feature. Since this is only supported in this path, running such models requires providing the following environment variables: `VLLM_USE_FLASHINFER_MOE_FP4=1 VLLM_FLASHINFER_MOE_BACKEND=throughput`
3. Fixing an issue with scale padding that is more common in non-gated models: If the intermediate size is not a multiple of 128, the swizzling logic pads the weight scales. However, for the flashinfer kernel to work the sizes of the weights and scales must match. Therefore, we added a logic to pad the weights of the up and down projections as well to match the padded size of the weight scales. This is more common in non-gated models since if the model is gated, we only require its intermediate size to be a multiple of 64, since there are 2 weights packed into `w13_weight_scale` in such a case.

## Test Plan
We've fixed and expanded the flashinfer fused moe test to also test a non-gated activation (relu2).

## Test Result
Old tests were fixed and new tests are passing internally.

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/flashinfer_cutlass_moe.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def workspace_shapes(
         - Note: in order for activation chunking to work, the first dimension
           of each tuple must be the number of tokens.
         """
+
```
Comment (tlrmchlsmth) [inline vllm/model_executor/layers/fused_moe/flashinfer_cutlass_moe.py]: nit: try to avoid whitespace-only changes in otherwise untouched files

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 2]
```diff
@@ -566,14 +566,20 @@ def _get_quant_method() -> FusedMoEMethodBase:
...
+                (
+                    UnquantizedFusedMoEMethod,
+                    ModelOptFp8MoEMethod,
+                    ModelOptNvFp4FusedMoE,
+                ),
             ):
                 raise NotImplementedError(
                     "is_act_and_mul=False is supported only for unquantized "
-                    "and ModelOpt FP8 moe for now"
+                    ", FP8 and modelopt FP4 checkpoints"
```
Comment (tlrmchlsmth) [inline vllm/model_executor/layers/fused_moe/layer.py]: for clarity
```suggestion
                    ", ModelOpt FP8, and ModelOpt NvFp4 checkpoints"
```

File: vllm/model_executor/layers/quantization/modelopt.py [hunk 3]
```diff
@@ -1586,11 +1592,39 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
...
 
+            w13_weight = layer.w13_weight
+            intermediate_size_pad = w13_blockscale_swizzled.size(1) - w13_weight.size(1)
+            if intermediate_size_pad:
+                # padding gated activations will require to split w1 and w3
+                # and pad them individually
+                assert not self.moe.is_act_and_mul, (
+                    "The intermediate size required padding, "
+                    "but padding is not implemented for non-gated activations"
+                )
```
[the remark this answers is not in the document]
Comment (tlrmchlsmth) [reply vllm/model_executor/layers/quantization/modelopt.py:1458]: please address the bot's comment

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 4]
```diff
@@ -1111,7 +1117,7 @@ def weight_loader(
                     self._load_combined_w13_weight_scale(
                         shard_dim=shard_dim,
                         loaded_weight=loaded_weight,
-                        param=param,
+                        param=param[expert_id],
                         tp_rank=self.tp_rank,
```
[the remark this answers is not in the document]
Comment (tlrmchlsmth) [reply vllm/model_executor/layers/fused_moe/layer.py:1287]: @omera-nv could you please check this?

File: vllm/model_executor/layers/fused_moe/flashinfer_cutlass_moe.py (hunk 1 above)
Comment (omera-nv) [reply vllm/model_executor/layers/fused_moe/flashinfer_cutlass_moe.py]: Fixed

File: vllm/model_executor/layers/fused_moe/layer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (omera-nv) [reply vllm/model_executor/layers/fused_moe/layer.py:1287]: `param` is correct when the experts are all in a single weight, `param[expert_id]` is for when experts are kept in separate weights. Seems like `expert_data` is always correct so I'll use that instead.

**Review by tlrmchlsmth — APPROVED** [review]
Looks good to me now, thanks!

**Comment by tlrmchlsmth** [comment]
Please fix the merge conflicts, and I'll mark the PR ready and turn on automerge

**Review by mgoin — APPROVED** [review]
